File: services/search-service/src/main/java/com/yoloo/server/search/user/UserController.java
package com.yoloo.server.search.user;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.solr.core.query.SolrPageRequest;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RequestMapping("/api/search")
@RestController
class UserController {

  private final UserRepository userRepository;

  @Autowired
  UserController(UserRepository userRepository) {
    this.userRepository = userRepository;
  }

  @GetMapping("/users")
  Iterable<User> searchUsers(@RequestParam(value = "q", required = false) String query) {
    return userRepository.findUsersByDisplayName(query, new SolrPageRequest(0, 30));
  }
}

File: src/main/kotlin/com/yoloo/server/core/user/application/UserController.kt
package com.yoloo.server.core.user.application

import com.googlecode.objectify.ObjectifyService.ofy
import com.yoloo.server.core.user.domain.model.*
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDate
import java.util.*

@RestController
@RequestMapping("/api/v1")
class UserController {

    @GetMapping("/users")
    fun getUser(): User {
        return ofy().load().type(User::class.java).first().now()
    }

    @GetMapping("/users2")
    fun createUser() : ResponseEntity<User> {
        val user = User(
            displayName = UserDisplayName(displayName = "Yasin Sinan Kayacan"),
            avatarUrl = "",
            locale = Locale.ENGLISH,
            userMeta = UserMeta(password = "", lastKnownIP = ""),
            email = Email("y.kayacan@emakina.com.tr"),
            scopes = setOf("user:read", "user:write"),
            birthDate = LocalDate.now(),
            countryIsoCode = "",
            subscribedGroups = listOf(UserGroup(id = 10, displayName = "group1"))
        )

        ofy().save().entity(user).now()

        return ResponseEntity.ok(user)
    }
}